### ISSUE TITLE
Tutorial Request - Extending the Plugin

### ISSUE THREAD

**Issue by cmwwebfx** [michaeluno/admin-page-framework#188]
In the latest Admin Page Framework plugin on WP repo, you have a nifty way of activating the default fieldset, and activating the extended fields in the 3rd part tools. I would really enjoy the read of how you achieved that, since it is a great way to make extensions for plugins built using the APF framework. 

Can we have these activate or download and install from an external repo? Also, is there a way to make small plugins that will extend a plugin further using APF?

Ciaran

**Comment by michaeluno** [comment]
Hi Ciaran,

> Can we have these activate or download and install from an external repo? 

Are you asking whether it is possible to activate a plugin hosted not on wordpress.org through a custom activation UI? If so, I'm not familiar with that method. Though I think it should be possible but it takes time for me to research how.

The demo components of the loader plugin is already bundled in the loader plugin so it's not something like it is downloaded from an external site and activated. 

> Also, is there a way to make small plugins that will extend a plugin further using APF?

Yes. The Field Type Pack add-on is like that. It extends the framework loader plugin.

**Comment by cmwwebfx** [comment]
Hmm, it is hard to describe since i don't know the right terms to use for this scenario.

But yes, when I install the plugin version (latest APF) from WP, I am greeted with a page that has 2 boxes on it. 1 box to activate the demo fields, and another box that goes to your website to purchase the addon pack. I assume that once a purchase is made then the framework is extended.

If this is a correct assumption on how this is extended, I would love to see a tutorial on how this is all achieved. It makes sense to build a plugin that can be extended through small purchases of extensions as a business model.

**Comment by michaeluno** [comment]
>  I assume that once a purchase is made then the framework is extended.

Ah, it does not automatically install it after a purchase. You'll download a plugin zip file when you purchase the add-on and you need to upload it via the WordPress plugin activation UI (Dashboard -> Plugins->Add New -> Upload). 

Though, is it what you assumed that after a purchase, you get redirected back to the Add-ons page and now it is installed and activated?

**Comment by cmwwebfx** [comment]
That is a good description for what I thought would happen. Purchase extension, download and upload via wp plugin upload, once activated it is redirected to that page where the other little box was to activate or already activated.

**Comment by michaeluno** [comment]
> Purchase extension, download and upload via wp plugin upload, once activated it is redirected to that page where the other little box was to activate or already activated. 

I'm a bit confused. I thought you did not assume you need to download and upload a file. Anyway, can you rephrase your request? I'm still not sure about what you'd like to know about or I may have already answered.

**Comment by cmwwebfx** [comment]
Sorry, it is due to me not knowing the correct terminology of this process.

You have a page that has 2 boxes on it. One that enables the demo fields, and one that directs the user to purchase and download the extension.

I will like to know in the tutorial how you made the page that has those 2 boxes. In the tutorial I will like to know how you have disabled the demo fields and once you press activate they all get loaded.

I will also like to know how you achieve the extension process of APF. ie, package up a small extension plugin, then upload to the site and have it as an extension of the main APF plugin.

This is not so much of an APF tutorial, rather a general usage tutorial since I find your tutorials are easy to follow since you go into great depth with your steps. And since you have already achieved this for APF with your extension and the page where to enable the demo fields, I thought it a great tutorial to ask for on how you did all that.

Again, I would think that others would find this incredibly helpful and appreciate the step by step guidance. For me I would pay to download the tutorial.

**Comment by michaeluno** [comment]
So you want to know how I built the framework loader plugin. Ha ha. I'll think about that. 

Anyway, I'm going to close this topic as it turned out to be not an issue of APF.  Apart form it, I think there should be a tutorial on how to properly load own plugin on top of the framework loader plugin. 

Thanks for sharing your thoughts!

**Comment by cmwwebfx** [comment]
Thank you Michael. I don't think I need to know how to build the framework loader unless it is not included into the framework itself for us to use? Hopefully my request for a tutorial was not out of reach for a tutorial of how we can make such a thing like you have in APF. If the loader plugin is not included in the framework, then possibly I am not sure what I am looking for.
